### DISCUSSION TITLE
When using turbopack on Next13, NEXT_REDIRECT error is thrown when redirect called from within an async function

### DISCUSSION THREAD

**Discussion by zdaar** [vercel/turborepo#5579] [Help] [answered]
**Describe the bug**
When using the `redirect` function in an async function in turbopack enabled Nextjs 13, a `NEXT_REDIRECT` error is thrown, as expected, to halt execution and initiate the redirect. However, I am puzzled as why this "error" is handled  differently between the standard Next.js compiler and Turbopack.

With the Webpack-based compiler, the `NEXT_REDIRECT` error is not shown in the output, which I assume is because it's expected behavior in the context of Next.js's flow control mechanisms.

On the other hand, when using Turbopack, these `NEXT_REDIRECT` errors are displayed both in the console and on the page. I suspect that Turbopack interprets them as unexpected exceptions due to its different handling of bundling and execution flow.

Should I file a bug report or am I missing something here ? Am I even in the right place ? Should I go to the Next Team for this ?

I tested this on next 13.4.4 and 13.4.10, React 18.2.0

**To Reproduce**
Here is a snippet which reproduces the issue:

```javascript
import { getServerSession } from "next-auth";
import { redirect } from "next/navigation";
import { authOptions } from "./api/auth/[...nextauth]/route";

export default async function Home() {
  const session = await getServerSession(authOptions);

  if (!session) {
    redirect("/api/auth/signin");
  }
}
```
![ShareX_17KP4KX8Te](https://github.com/vercel/turbo/assets/4218386/1ad14321-f176-44c9-b6ad-4935d9333be4)

**Comment by anthonyshew** [comment, marked as answer]
Please open an issue if you're still seeing this problem. 🙏
